plugins/verifier/vulnerabilityreport/vulnerability_report.go

```diff
@@ -0,0 +1,400 @@
+	SeverityRegex               = `Severity:\s*(\w+)`
+)
+
+type PluginConfig struct {
+	Name               string   `json:"name"`
+	SchemaURL          string   `json:"schemaURL,omitempty"`
+	MaximumAge         string   `json:"maximumAge,omitempty"`
+	DisallowedSeverity []string `json:"disallowedSeverity,omitempty"`
+	Passthrough        bool     `json:"passthrough,omitempty"`
+	DenylistCVEs       []string `json:"denylistCVEs,omitempty"`
+}
+
+type PluginInputConfig struct {
+	Config PluginConfig `json:"config"`
+}
+
+func main() {
+	skel.PluginMain("vulnerabilityreport", "1.0.0", VerifyReference, []string{"1.0.0"})
+}
+
+func parseInput(stdin []byte) (*PluginConfig, error) {
+	conf := PluginInputConfig{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse stdin for the input: %w", err)
+	}
+
+	return &conf.Config, nil
+}
+
+func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, referenceDescriptor ocispecs.ReferenceDescriptor, referrerStore referrerstore.ReferrerStore) (*verifier.VerifierResult, error) {
+	input, err := parseInput(args.StdinData)
+	if err != nil {
+		return nil, err
+	}
+
+	// check report is newer than allowed maximum age
+	if input.MaximumAge != "" {
+		ok, created, err := validateMaximumAge(input.MaximumAge, referenceDescriptor)
+		if err != nil {
+			return &verifier.VerifierResult{
+				Name:      input.Name,
+				IsSuccess: false,
+				Message:   fmt.Sprintf("vulnerability report validation failed: error validating maximum age:[%v]", err.Error()),
+			}, nil
+		}
+		if !ok {
+			return &verifier.VerifierResult{
+				Name:      input.Name,
+				IsSuccess: false,
+				Message:   fmt.Sprintf("vulnerability report validation failed: report is older than maximum age:[%s]", input.MaximumAge),
+				Extensions: map[string]interface{}{
+					CreatedAnnotation: created,
+				},
+			}, nil
+		}
+	}
+
+	ctx := context.Background()
+
+	referenceManifest, err := referrerStore.GetReferenceManifest(ctx, subjectReference, referenceDescriptor)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching reference manifest for subject: %s reference descriptor: %v", subjectReference, referenceDescriptor.Descriptor)
+	}
+
+	if len(referenceManifest.Blobs) == 0 {
+		return &verifier.VerifierResult{
+			Name:      input.Name,
+			IsSuccess: false,
+			Message:   fmt.Sprintf("vulnerability report validation failed: no blobs found for referrer %s@%s", subjectReference.Path, referenceDescriptor.Digest.String()),
+		}, nil
+	}
+
+	blobDesc := referenceManifest.Blobs[0]
+	refBlob, err := referrerStore.GetBlobContent(ctx, subjectReference, blobDesc.Digest)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching blob for subject:[%s] digest:[%s]", subjectReference, blobDesc.Digest)
+	}
+
+	// skip all validation if passthrough is enabled
+	if input.Passthrough {
+		return &verifier.VerifierResult{
+			Name:      input.Name,
+			IsSuccess: true,
```

Comment:
wonder if this should be true/false in passthrough mode.

Reply:
I think it should be true here. in my opinion, the `isSuccess` value in passthrough mode denotes if the manifest/blob fetch operations were successful. Open to counter points though...